scripts/scripts.js

```diff
@@ -137,13 +137,66 @@ export async function loadTemplate(doc, templateName) {
   }
 }
 
+function handleTopBanner(topBanner) {
+  if (topBanner && !topBanner.classList.contains('dismissed')) {
+    const header = document.querySelector('header');
+    const bannerHeight = topBanner.offsetHeight;
+
+    document.body.classList.add('has-top-banner');
+    document.body.style.paddingTop = `${bannerHeight}px`;
+
+    if (header) {
+      header.style.top = `${bannerHeight}px`;
+    }
+  } else {
+    document.body.classList.remove('has-top-banner');
+    document.body.style.paddingTop = '0';
+
+    const header = document.querySelector('header');
+    if (header) {
+      header.style.top = '0';
+    }
+  }
+}
+
+function setupTopBannerObserver() {
+  const header = document.querySelector('header');
+
+  // watch for changes in the header
+  const observer = new MutationObserver((mutations) => {
+    mutations.forEach((mutation) => {
+      if (mutation.type === 'childList') {
+        const topBanner = document.querySelector('.top-banner');
+        if (topBanner) {
+          handleTopBanner(topBanner);
+          observer.disconnect(); // Stop observing once we've handled the banner
+        }
+      }
+    });
+  });
+
+  // observing the header for changes
+  observer.observe(header, { childList: true, subtree: true });
+
+  // listen for the custom event
+  window.addEventListener('topbannerloaded', () => {
+    const topBanner = document.querySelector('.top-banner');
+    if (topBanner) {
+      handleTopBanner(topBanner);
+    }
+  }, { once: true });
+}
+
 async function loadTopBanner(doc) {
   const topBannerFragment = await loadFragment('/fragments/top-banner');
   if (topBannerFragment) {
     const topBanner = topBannerFragment.querySelector('.top-banner');
     if (topBanner) {
       const header = doc.querySelector('header');
       header?.prepend(topBanner);
+
+      // Dispatch a custom event when the banner is added
+      window.dispatchEvent(new CustomEvent('topbannerloaded'));
```

Comment:
I feel like non of this code is required if we just allow the dom to reposition itself.

Reply:
I had to do this because of the way the site is already implemented. This is what happens when I allow it to reposition itself. This will require a lot of refactoring work to accomplish and we can do that in a different task.

![Screenshot 2024-07-16 at 11 08 56 AM](https://github.com/user-attachments/assets/d5221983-abf5-408b-9389-b3497486ff7e)